cosmo_tester/test_suites/cluster/full_cluster_test.py

```diff
@@ -398,35 +407,46 @@ def _verify_status_when_rabbit_inactive(broker1, broker2, broker3, logger,
                                         client):
     logger.info('Stopping one of the rabbit nodes')
     broker1.run_command('systemctl stop cloudify-rabbitmq', use_sudo=True)
-    time.sleep(10)
 
+    _validate_status_when_one_rabbit_inactive(broker1, logger, client)
+
+    logger.info('Stopping the other rabbit nodes')
+    broker2.run_command('systemctl stop cloudify-rabbitmq', use_sudo=True)
+    broker3.run_command('systemctl stop cloudify-rabbitmq', use_sudo=True)
+
+    _validate_status_when_all_rabbits_inactive(logger, client)
+
+
+# It can take time for prometheus state to update.
+# Thirty seconds should be much more than enough.
+@retrying.retry(stop_max_attempt_number=15, wait_fixed=2000)
+def _validate_status_when_one_rabbit_inactive(broker, logger, client):
+    logger.info('Checking status reporter with one rabbit down...')
     cluster_status = client.cluster_status.get_status()
     broker_service = cluster_status['services']['broker']
     assert cluster_status['status'] == ServiceStatus.DEGRADED
     assert broker_service['status'] == ServiceStatus.DEGRADED
-    assert broker_service['nodes'][broker1.hostname]['status'] == \
+    assert broker_service['nodes'][broker.hostname]['status'] == \
         ServiceStatus.FAIL
     manager_status = client.manager.get_status()
     assert manager_status['status'] == ServiceStatus.HEALTHY
 
-    logger.info('Stopping the other rabbit nodes')
-    broker2.run_command('systemctl stop cloudify-rabbitmq', use_sudo=True)
-    broker3.run_command('systemctl stop cloudify-rabbitmq', use_sudo=True)
-    time.sleep(10)
 
+# It can take time for prometheus state to update.
+# Thirty seconds should be much more than enough.
+@retrying.retry(stop_max_attempt_number=15, wait_fixed=2000)
+def _validate_status_when_all_rabbits_inactive(logger, client):
+    logger.info('Checking status reporter with all syncthing down...')
```

Comment:
I guess it's not about syncthing, or is it?

Reply:
```suggestion
    logger.info('Checking status reporter with all rabbits down...')
```

Reply:
/Everything/ is about syncthing.

Except this, and some other things. Good catch.

Reply:
Updated.